test.js

```diff
@@ -0,0 +1,38 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+var TestUtils = require('react-addons-test-utils');
+var OnClickOutside = require('.');
+
+describe('react-onclickoutside', function() {
+
+  var OutsideClickComponent = React.createClass({
+    mixins: [OnClickOutside],
+
+    getInitialState: function() {
+      return {handleClickOutsideCalled: false};
+    },
+
+    handleClickOutside: function() {
+      this.setState({handleClickOutsideCalled: true});
+    },
+
+    render: function() {
+      return React.createElement('div');
+    }
+  });
+
+  it('should not call handleClickOutside when clicking the element', function() {
+    var element = React.createElement(OutsideClickComponent);
+    var component = TestUtils.renderIntoDocument(element);
+    ReactDOM.findDOMNode(component).dispatchEvent(new MouseEvent('mousedown'));
+    assert(!component.state.handleClickOutsideCalled);
+  });
+
+  it('should call handleClickOutside when clicking the document', function() {
+    var element = React.createElement(OutsideClickComponent);
+    var component = TestUtils.renderIntoDocument(element);
+    document.dispatchEvent(new MouseEvent('mousedown'));
```

Comment:
I'm sort or taking the mindset that works in Chrome, Firefox, and IE == works on the internet, and is slightly more realistic than phantomjs.  Haven't added the others yet because this is a WIP. =P  Haven't experimented with phantom here yet, but if that's also equivalent to "working on the internet" in your eyes then we should try that out.